cellprofiler/gui/figure.py

```diff
@@ -1593,19 +1604,35 @@ def subplot_imshow_labels(
                      consistent label colors in multiple displays)
         :return:
         """
-        if background_image is None:
-            background_image = numpy.zeros_like(image, dtype=numpy.float32)
-            opacity = 1.0
-        else:
+        if background_image is not None:
             opacity = 0.7
-
-        label_image = cellprofiler.object.overlay_labels(
-            labels=image,
-            opacity=opacity,
-            pixel_data=background_image,
-            max_label=max_label,
-            seed=seed
-        )
+            label_image = cellprofiler.object.overlay_labels(
+                labels=image,
+                opacity=opacity,
+                pixel_data=background_image,
+                max_label=max_label,
+                seed=seed
+            )
+            colormap = None
+        else:
+            # Mask the original labels
```

Comment:
We could consider breaking this into a function to emphasize the underlying logic.